File: tests/fixtures/code_samples/sample.lua
-- Sample Lua module for testing code parsing

local M = {}

function M.add(a, b)
    return a + b
end

local function helper(x)
    if x > 0 then
        return x * 2
    else
        return 0
    end
end

function M.process(items)
    local result = 0
    for i, v in ipairs(items) do
        if v > 0 then
            result = result + helper(v)
        end
    end
    return result
end

return M

File: tests/fixtures/code_samples/sample.swift
/// Sample Swift code for testing code parsing

/// Protocol for shapes
protocol Shape {
    func area() -> Double
}

/// Rectangle struct
struct Rectangle: Shape {
    let width: Double
    let height: Double

    /// Calculate area
    func area() -> Double {
        return width * height
    }
}

/// Circle class
class Circle: Shape {
    let radius: Double

    /// Initialize with radius
    init(radius: Double) {
        self.radius = radius
    }

    /// Calculate area
    func area() -> Double {
        return 3.14159 * radius * radius
    }
}
